addons/account/models/account_journal.py

```diff
@@ -474,6 +479,10 @@ def _fill_missing_values(self, vals):
         if 'refund_sequence' not in vals:
             vals['refund_sequence'] = vals['type'] in ('sale', 'purchase')
 
+        # === Fill missing alias name ===
+        if journal_type in ('sale', 'purchase') and 'alias_name' not in vals:
+            vals['alias_name'] = '%s.%s' % (company.name, vals.get('code'))
```

Comment:
`alias_name` is a `related` field... I'm not sure it works like that. The whole management of alias are more magical. See https://github.com/odoo/odoo/blob/8416b236893c3ce7258e02fc9ad04a81bbb65af8/addons/account/models/account_journal.py#L336

Reply:
It works because the _update_mail_alias is called by the create() method later. 

https://github.com/odoo/odoo/blob/8416b236893c3ce7258e02fc9ad04a81bbb65af8/addons/account/models/account_journal.py#L486-L494